### ISSUE TITLE
icinga-php-library (>=0.9.0) required by icinga db but none available

### ISSUE THREAD

**Issue by wrufeger** [Icinga/icinga-php-library#11]
can only find 0.8.1 and nightly snapshot (which btw. won't work very well)

**Comment by nilmerg** [comment]
> (which btw. won't work very well)

What does that mean? You can only use the snapshot, so it'd be very useful to know what doesn't work.

**Comment by wrufeger** [comment]
With snapshot I get the following error in icinga:
`
Uncaught ArgumentCountError: Too few arguments to function ipl\Orm\Resolver::__construct(), 0 passed in /usr/share/icingaweb2/modules/icingadb/library/Icingadb/Redis/VolatileStateResults.php on line 62 and exactly 1 expected in /usr/share/icinga-php/ipl/vendor/ipl/orm/src/Resolver.php:55
Stack trace:
#0 /usr/share/icingaweb2/modules/icingadb/library/Icingadb/Redis/VolatileStateResults.php(62): ipl\Orm\Resolver->__construct()
#1 /usr/share/icingaweb2/modules/icingadb/library/Icingadb/Redis/VolatileStateResults.php(46): Icinga\Module\Icingadb\Redis\VolatileStateResults->applyRedisUpdates()
#2 /usr/share/icingaweb2/modules/icingadb/library/Icingadb/Common/BaseItemList.php(87): Icinga\Module\Icingadb\Redis\VolatileStateResults->rewind()
#3 /usr/share/icingaweb2/modules/icingadb/library/Icingadb/Widget/ItemList/StateList.php(20): Icinga\Module\Icingadb\Common\BaseItemList->assemble()
#4 /usr/share/icinga-php/ipl/vendor/ipl/html/src/HtmlDocument.php(295): Icinga\Module\Icingadb\Widget\ItemList\StateList->assemble()
`

<img width="563" alt="image" src="https://user-images.githubusercontent.com/524191/167100775-100aea13-1b73-412a-9fe8-a6946b2d17af.png">

**Comment by nilmerg** [comment]
Then Icinga DB Web is outdated. Recent icinga-php-library snapshot and recent Icinga DB Web master works.

**Comment by wrufeger** [comment]
installed 1.0.0-rc2 from icinga debian repository. So which one I should use ?

**Comment by nilmerg** [comment]
You came here because Icinga DB Web asks for icinga-php-library in version 0.9.0. This can only be the case if you're using the master git branch of it. None of our packages for Icinga DB Web are recent enough. So I guess, pulling the master branch should solve your problem.

**Comment by wrufeger** [comment]
Then probably the debian repositories are outdated mentioned at https://icinga.com/docs/icinga-db/latest/doc/02-Installation/. icinga-bullseye-testing (last updated 2022/02/01) contains rc1, icinga-bullseye (2022/04/28) contains rc2 and icinga-bullseye-snapshots messed up whole installation.

Now installed master branch from scratch and snapshot branch from icinga-php-library, but still problems:.

`Fatal error
: Declaration of Icinga\Module\Director\Dashboard\Dashboard::count() must be compatible with ipl\Html\HtmlDocument::count(): int in
/usr/share/icingaweb2/modules/director/library/Director/Dashboard/Dashboard.php
on line 152`

seems to be related to this project

**Comment by nilmerg** [comment]
I really don't know what's going on here, but your installation seems to be messed up somehow. The error you'll get in the Director can only happen if you have a version of icinga-php-library that's not been released yet (e.g. snapshot/nightly), but is still outdated. The recent build of snapshot/nightly shouldn't show the same error.

Yet, Icinga DB Web is obviously outdated (Doesn't work with snapshot/nightly, whether recent or not) but you think it requires version 0.9.0 of icinga-php-library. How did you came to this conclusion? Did the packages install it automatically? Then, I'm afraid, you should keep away from the packages as they're then broken.

Though, I wouldn't recommend the packages anyway. They don't represent the current state of development and are heavily outdated. The *repository* `icinga-bullseye-testing` may have been updated at 2022/02/01, but the packages were definitely not. They're from November last year.

What I'd recommend is to directly check out the git master of Icinga DB and Web as well as the `snapshot/nightly` branch of icinga-php-library and icinga-php-thirdparty. Or to use the icinga docker images. (The `:master` tag of course)

**Comment by wrufeger** [comment]
Thats what I did before getting the last Director related error. Removed debian packages, installed master from Icinga DB (compiled from source with go), Icinga DB Web and snapshots from ipl and third-party:

- Icinga 2 | 2.13.3-1
- Icingaweb 2 | 2.10.1
- Icinga DB | v1.0.0-rc2-dev
- Icinga DB Web | 1.0.0-rc2
- Director | 1.9.1 / master
- icinga-php-thirdparty | 0.11.0-dev
- icinga-php-library | 0.9.0-dev

running all on Debian 11 with PHP 7.4.28.

with ipl 0.8.1 director works, but errors on other pages. with ipl snapshot/nightly all works except director.
